### PR TITLE
HDDS-8320. Remove exclusion of jackson from ranger-intg

### PR DESCRIPTION
**Pull request by smengcl**
## What changes were proposed in this pull request?

It turns out excluding jackson in HDDS-7924 might have broken actual ranger-intg client usage:

```
E           23/03/28 20:05:54 INFO rpc.RpcClient: Creating Tenant: 'tenant303241680030324', with new volume: 'tenant303241680030324'
E           	at org.apache.ranger.plugin.util.RangerRESTClient.buildClient(RangerRESTClient.java:232)
E           	at org.apache.ranger.plugin.util.RangerRESTClient.getClient(RangerRESTClient.java:215)
E           	at org.apache.ranger.plugin.util.RangerRESTClient.post(RangerRESTClient.java:576)
E           	at org.apache.ranger.RangerClient.invokeREST(RangerClient.java:422)
E           	at org.apache.ranger.RangerClient.lambda$responseHandler$0(RangerClient.java:464)
E           	at org.apache.ranger.RangerClient.responseHandler(RangerClient.java:462)
E           	at org.apache.ranger.RangerClient.callAPI(RangerClient.java:527)
E           	at org.apache.ranger.RangerClient.createRole(RangerClient.java:334)
E           Caused by: java.lang.ClassNotFoundException: org.codehaus.jackson.jaxrs.JacksonJsonProvider
E           	... 26 more
```

Unfortunately our acceptance tests didn't cover the actual Ranger client testing because the infra is not there yet. iirc @djordje-mijatovic is trying to improve the docker dev env to spin up actual Ranger, at which point we could write new acceptance tests that uses ranger-intg client to run against actual Ranger Admin Server so that we can avoid such issues in the future.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8320

## How was this patch tested?

- Pending internal system tests to pass.
- New acceptance tests to be added after Ranger docker dev env is ready. Filed HDDS-8321 for it.

### CODE REVIEW

**Review by kerneltime — APPROVED** [review]

**Comment by smengcl** [comment]
Marking this as draft as @swamirishi is still tuning the Maven POM.

Will add LICENSE later.

File: hadoop-ozone/ozone-manager/pom.xml [hunk 1]
```diff
@@ -157,6 +157,13 @@ https://maven.apache.org/xsd/maven-4.0.0.xsd">
       <artifactId>jersey-client</artifactId>
     </dependency>
 
+    <!-- Overriding ranger-intg jackson version -->
+    <dependency>
+      <groupId>org.codehaus.jackson</groupId>
+      <artifactId>jackson-core-asl</artifactId>
+      <version>${jackson1.version}</version>
+    </dependency>
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/pom.xml:165]: This is what Ranger brings without the exclusion:

```
[INFO] |     +- org.codehaus.jackson:jackson-jaxrs:jar:1.9.13:compile
[INFO] |     |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
[INFO] |     |  \- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
```

If we want to add explicit dependency, I think we should add `jackson-jaxrs`, not `jackson-core-asl`.

**Review by adoroszlai — COMMENTED** [review]
Thanks @smengcl for the patch and sorry for the trouble caused by the exclusion.

File: hadoop-ozone/ozone-manager/pom.xml (hunk 1 above)
Comment (smengcl) [reply hadoop-ozone/ozone-manager/pom.xml:165]: Thanks @adoroszlai for looking at this. @swamirishi has added jaxrs in #4504 .

**Comment by smengcl** [comment]
Closing this one in favor of #4504 .

File: hadoop-ozone/ozone-manager/pom.xml (hunk 1 above)
Comment (swamirishi) [reply hadoop-ozone/ozone-manager/pom.xml:165]: Please check https://github.com/apache/ozone/pull/4504 for the fix.

**Pull request closed without merging**